### PR TITLE
feat(integration-template-tests): [nan-1696] update sdk so that record responses is less instrusive

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Use an axios interceptor to avoid adding logic to the SDK and so that the logic for saving down responses is in one place. This was suggested in https://github.com/NangoHQ/nango/pull/2678#discussion_r1747576571 and discussed further in Slack.

## Issue ticket number and link
NAN-1696

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/sdk/sync.ts [hunk 1]
```diff
@@ -424,6 +433,14 @@ export class NangoAction {
         if (config.dryRun) {
             this.dryRun = config.dryRun;
             this.saveResponses = config.saveResponses || false;
```
Comment (bodinsamuel) [inline packages/shared/lib/sdk/sync.ts]: Is `saveResponses` still useful?

**Review by bodinsamuel — APPROVED** [review]
looks good

File: packages/shared/lib/sdk/sync.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/sdk/sync.ts]: Nope! Thanks 🙏

Updated with https://github.com/NangoHQ/nango/pull/2763/commits/4cd4273bcd20a91759261ec9f9bbb0a377fb0c93

File: packages/cli/lib/services/response-saver.service.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ensureDirectoryExists(`${directoryName}/mocks`);
+
+    let filePath: string;
+    if (customFilePath) {
+        filePath = path.join(directoryName, customFilePath);
+        ensureDirectoryExists(path.dirname(filePath));
+    } else {
+        const fileName = config.endpoint.replace('/', '');
+        filePath = `${directoryName}/mocks/${config.method?.toLowerCase()}/${fileName}.json`;
+        ensureDirectoryExists(path.dirname(filePath));
```
Comment (TBonnin) [inline packages/cli/lib/services/response-saver.service.ts:33]: you can put this outside of the ifelse

File: packages/node-client/lib/index.ts [hunk 3]
```diff
@@ -107,6 +117,12 @@ export class Nango {
                 'User-Agent': this.userAgent
             }
         });
+
+        if (interceptors) {
+            if (interceptors.response) {
```
Comment (TBonnin) [inline packages/node-client/lib/index.ts:122]: `if(interceptors?.response) {`

File: packages/shared/lib/sdk/sync.ts [hunk 4]
```diff
@@ -613,15 +619,6 @@ export class NangoAction {
...
             this.memoizedConnections.set(credentialsPair, { connection, timestamp: Date.now() });
-            if (this.saveResponses) {
-                const directoryName = `${process.env['NANGO_MOCKS_RESPONSE_DIRECTORY']}${providerConfigKey}`;
-                responseSaver.saveResponse<Pick<Connection, 'metadata' | 'connection_config'>>({
-                    directoryName,
-                    config: { endpoint: 'getConnection', providerConfigKey } as ProxyConfiguration,
-                    data: { metadata: connection.metadata as Metadata, connection_config: connection.connection_config },
-                    customFilePath: 'mocks/nango/getConnection.json'
-                });
-            }
```
Comment (TBonnin) [inline packages/shared/lib/sdk/sync.ts:624]: ❤️

**Review by TBonnin — COMMENTED** [review]
Much cleaner. very good job

**Pull request merged**